### PR TITLE
PD-303: Port <RadioBoxGroup> to TypeScript

### PR DESCRIPTION
**Pull request by jetpacmonkey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-303: Port <RadioBoxGroup> to TypeScript](https://jira.mongodb.org/browse/PD-303)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/RadioBoxGroup/src/RadioBox.tsx [hunk 1]
```diff
@@ -138,23 +151,35 @@ export const radioWrapper = css`
   }
 `;
 
-const getStatefulStyles = states => ({
+const getStatefulStyles = (states: StateForStyles) => ({
```
Comment (DesignerDave) [inline packages/RadioBoxGroup/src/RadioBox.tsx]: For consistency's sake, probably best to do `states` -> `state`

File: packages/RadioBoxGroup/src/RadioBox.tsx [hunk 2]
```diff
@@ -7,7 +7,15 @@ const { css, cx } = emotion;
...
 const radioBoxInput = createDataProp('radio-box-input');
 
-export const radioBoxSizes = {
+export const Size = {
+  Default: 'default',
+  Compact: 'compact',
+  Full: 'full',
+} as const;
+
+export type SizeValues = typeof Size[keyof typeof Size];
```
Comment (DesignerDave) [inline packages/RadioBoxGroup/src/RadioBox.tsx]: You sorta established a pattern for these shared props in the RadioGroup PR – should we move these to a separate file for each component to import from?

File: packages/RadioBoxGroup/src/RadioBoxGroup.tsx [hunk 3]
```diff
@@ -1,15 +1,42 @@
...
+}
+
+interface RadioBoxGroupState {
+  value: string | number;
+}
+
+function isRadioBoxElement(
+  element: React.ReactNode,
+): element is React.ReactElement<RadioBoxProps, typeof RadioBox> {
+  return (
```
Comment (DesignerDave) [inline packages/RadioBoxGroup/src/RadioBoxGroup.tsx]: I commented on this in the other PR as well – any reason we can't do the following?

```js
return element && element.type && element.type.displayName === 'RadioBox'
```

File: packages/RadioBoxGroup/src/RadioBox.tsx (hunk 2 above)
Comment (jetpacmonkey) [reply packages/RadioBoxGroup/src/RadioBox.tsx]: yup, Harry hadn't told me about the same-name fancy magic thingy

File: packages/RadioBoxGroup/src/RadioBoxGroup.tsx (hunk 3 above)
Comment (jetpacmonkey) [reply packages/RadioBoxGroup/src/RadioBoxGroup.tsx]: Discussed on the other PR. Resolving.

File: packages/RadioBoxGroup/src/RadioBox.tsx (hunk 2 above)
Comment (jetpacmonkey) [reply packages/RadioBoxGroup/src/RadioBox.tsx]: Oh I misread the comment. Separate file would kinda make sense here...

**Pull request merged**